src/pages/Profile/UserSettings/UserSettings.js

```diff
@@ -38,14 +45,57 @@ class UserSettings extends Component {
 
   /** Save the latest user settings modified by the user */
   saveUserSettings = _debounce((settings) => {
+    // We cannot save the user settings if the basemap names are not unique.
+    // Otherwise, this will mess up our attempts to match the server generated
+    // id with the matching custom basemap with that name.
+    if (!this.areBasemapNamesUnique(settings.customBasemaps)) {
+      return
+    }
+
     this.setState({isSaving: true, saveComplete: false})
 
-    const editableUser = AsEditableUser(settings)
-    editableUser.normalizeDefaultBasemap()
+    const editableUser = AsEditableUser(_cloneDeep(settings))
+
+    // Massage customBasemaps json:
+    // 1.don't send if name or url has not been completed yet
+    // 2. insert id: -1 if it's a new customBasemap
+    _remove(editableUser.customBasemaps, (data) => (!data.name || !data.url))
```

Comment:
Would `_isEmpty` (perhaps after a `_trim`) be safer here in case the name or url is merely whitespace?